Makefile

```diff
@@ -29,6 +29,116 @@ clean: pykwasm
 	$(KDIST) clean
 
 
+# Building ULM-integrated Definition
+# ----------------------------------
+
+ULM_BUILD_DIR=./build
+ULM_LIB_DIR=$(ULM_BUILD_DIR)/lib
+ULM_DEP_DIR=$(ULM_BUILD_DIR)/deps
+ULM_CXX=$(shell [ $(origin CXX) != default ] && echo $(CXX) )
+ULM_KF_INCLUDE_DIR=$(shell dirname "`which llvm-kompile`")/../include
+
+ULM_WASM_DIR=$(ULM_BUILD_DIR)/wasm
+ULM_WASM_MAIN=pykwasm/src/pykwasm/kdist/wasm-semantics/ulm-wasm.md
+ULM_WASM_LIB=libkwasm.so
+ULM_WASM_TARGET=$(ULM_LIB_DIR)/$(ULM_WASM_LIB)
+
+ULM_WASM_COMPILER_TARGET=$(ULM_BUILD_DIR)/ulm-contract-compiler
+
+ULM_PLUGIN_DIR=$(ULM_DEP_DIR)/plugin
+ULM_PLUGIN_LIB=krypto.a
+ULM_PLUGIN_TARGET=$(ULM_LIB_DIR)/$(ULM_PLUGIN_LIB)
+
+ULM_HOOKS_CLONE_DIR=$(ULM_DEP_DIR)/ulm
+ULM_HOOKS_DIR=$(ULM_HOOKS_CLONE_DIR)/kllvm
+ULM_HOOKS_SRC=ulm_kllvm.cpp ulm_hooks.cpp ulm_kllvm_c.cpp
+ULM_HOOKS_LIB=libulmkllvm.so
+ULM_HOOKS_TARGET=$(ULM_LIB_DIR)/$(ULM_HOOKS_LIB)
+
+### ULM Plugin
+
+$(ULM_PLUGIN_DIR)/.git:
```

Comment:
I'm not sure if it's worth renaming, but I usually think of this plugin as "the crypto plugin", not "the ulm plugin".

Reply:
I updated these makefile target variable names to use the name "krypto."